e2e/e2e_test.go

```diff
@@ -121,79 +118,58 @@ func Test_E2E_Only_Address_Memo(t *testing.T) {
 	// Step 2 - Verify the submitted topic messages
 	ethTransactionHash, receivedSignatures := verifyTopicMessages(setupEnv, transactionResponse, 0, t)
 
-	// Step 3 - Verify Transaction Record in the database
-	expectedTxRecord := verifyTransferRecordInDB(
-		setupEnv.Clients.RouterContract,
-		setupEnv.DBVerifiers,
+	expectedTxRecord := prepareExpectedTransfer(
+		setupEnv,
 		transactionResponse.TransactionID,
 		"HBAR",
 		"0",
 		"0",
-		statusInfo{
-			status:          entity_transfer.StatusCompleted,
-			statusSignature: entity_transfer.StatusSignatureMined,
+		database.ExpectedStatuses{
+			Status:          entity_transfer.StatusCompleted,
+			StatusSignature: entity_transfer.StatusSignatureMined,
 		},
 		ethTransactionHash,
 		false, t)
 
-	// Step 4 - Verify Message Record in the database
-	verifyMessageRecordsInDB(setupEnv.DBVerifiers, expectedTxRecord, receivedSignatures, t)
+	// Step 4 - Verify Database Records
+	verifyDatabaseRecords(setupEnv, expectedTxRecord, receivedSignatures, t)
 }
 
-func verifyMessageRecordsInDB(DBVerifiers []*database.Service, record *entity.Transfer, signatures []model.SigDuplet, t *testing.T) {
-	for _, dbVerifier := range DBVerifiers {
-		exist, expectedMessageRecords, err := dbVerifier.SignatureMessagesExist(record, signatures)
-		if err != nil {
-			t.Fatalf("Could not figure out if messages for [%s] are the expected messages [%v] - Error: [%s].", record.TransactionID, expectedMessageRecords, err)
-		}
-		if !exist {
-			t.Fatalf("Messages for [%s] are not [%v].", record.TransactionID, expectedMessageRecords)
-		}
+func verifyDatabaseRecords(setupEnv *setup.Setup, expectedRecord *entity.Transfer, signatures []string, t *testing.T) {
+	exist, err := setupEnv.DbValidation.VerifyDatabaseRecords(expectedRecord, signatures)
+	if err != nil {
+		t.Fatalf("[%s] - Verification of database records failed - Error: [%s].", expectedRecord.TransactionID, err)
+	}
+	if !exist {
+		t.Fatalf("[%s] - Database does not contain expected records", expectedRecord.TransactionID)
 	}
 }
 
-type statusInfo struct {
-	status          string
-	statusSignature string
-	statusEthTx     string
-	statusEthTxMsg  string
-}
-
-func verifyTransferRecordInDB(routerContract *router.Router, DBVerifiers []*database.Service, transactionID hedera.TransactionID, nativeToken, txFee, gasPrice string, statuses statusInfo, ethTransactionHash string, shouldExecuteEthTx bool, t *testing.T) *entity.Transfer {
+func prepareExpectedTransfer(setupEnv *setup.Setup, transactionID hedera.TransactionID, nativeToken, txFee, gasPriceWei string, statuses database.ExpectedStatuses, ethTransactionHash string, shouldExecuteEthTx bool, t *testing.T) *entity.Transfer {
```

Comment:
nit: maybe we can pass only the `setupEnv.Clients.RouterContract` instead of the whole `setup.Setup` prop